p2p/host/relay/autorelay.go

```diff
@@ -196,16 +191,13 @@ func (ar *AutoRelay) tryRelay(ctx context.Context, pi peer.AddrInfo) bool {
 		return false
 	}
 
-	ok, err := circuit.CanHop(ctx, ar.host, pi.ID)
+	rsv, err := circuitv2client.Reserve(ctx, ar.host, pi)
 	if err != nil {
-		log.Debugf("error querying relay: %s", err.Error())
+		log.Debugf("error reserving slot on relay: %s", err.Error())
 		return false
 	}
 
-	if !ok {
-		// not a hop relay
-		return false
-	}
+	ar.host.Peerstore().AddAddrs(pi.ID, rsv.Relay.Addrs, peerstore.ConnectedAddrTTL)
```

Comment:
no, this has changed in the v2 code.
The addresses that you get are your own relay addresses, and they are in the `rsv(p).Addrs` field.

Reply:
You also need to schedule reservation refresh, use the `Expiration` field in the reservation to make the determination.

Reply:
done.